userspace/libsinsp/parsers.cpp

```diff
@@ -3673,7 +3673,7 @@ inline void sinsp_parser::process_recvmsg_ancillary_data_fds(scap_platform *scap
                                                              int const *fds,
                                                              size_t const fds_len,
                                                              scap_threadinfo *scap_tinfo) {
-	char error[SCAP_LASTERR_SIZE];
+	char error[SCAP_LASTERR_SIZE] = {};
```

Comment:
I don't think you need the extra assignment here, though the compiler probably gets rid of it anyways
```suggestion
	char error[SCAP_LASTERR_SIZE]{};
```

Reply:
That feels black magic syntax to me :O

Reply:
It's explicitly calling the default constructor of the `char[SCAP_LASTERR_SIZE]` type, initializing its contents to zero. The behavior when leaving that out is the same as it would be on C, just leave any garbage in the stack laying around.

There's a better explanation [here](https://stackoverflow.com/a/18295840)

Reply:
Again, the compiler probably catches that you are immediately copying new value into the array with the assignment and might just call the constructor for you, so either syntax will probably have the same result.